### DISCUSSION TITLE
Fixed periodicity

### DISCUSSION THREAD

**Discussion by adamjstewart** [cornellius-gp/gpytorch#2062] [Q&A] [answered]
I'm working with weather data, which I know will exhibit approximately sinusoidal trends with a periodicity of exactly 24 hours and 365 days. I would like to fix the periodicity of `PeriodicKernel` so that I only need to learn the lengthscale. In scikit-learn, this corresponds to:
```python
ExpSineSquared(periodicity=24, periodicity_bounds="fixed")
```
Is it possible to do this in GPyTorch? I tried:
```python
PeriodicKernel(period_length_constraint=Interval(24, 24, initial_value=24))
```
but it seems like this isn't allowed:
```
  File "/Users/ajstewart/.spack/.spack-env/view/lib/python3.9/site-packages/gpytorch/constraints/constraints.py", line 30, in __init__
    raise RuntimeError("Got parameter bounds with empty intervals.")
RuntimeError: Got parameter bounds with empty intervals.
```

**Comment by Balandat** [comment, marked as answer]
You could just implement a new kernel that uses the `forward` logic from `PeriodicKernel` but gets rid of the period as a learnable parameter:

```
class FixedPeriodPeriodicKernel(Kernel):

    def __init__(
        self,
        period_length: Tensor,
        **kwargs,
    ):
        super().__init__(**kwargs)
        self.period_length = period_length

    def forward(self, x1, x2, diag=False, **params):
        return PeriodicKernel.forward(self, x1, x2, diag=False, **params)
```

**Reply by adamjstewart** [reply]
Thanks, this looks promising, I'll try this out!

**Reply by adamjstewart** [reply]
Also had to add `has_lengthscale = True`, with that it works like a charm!
